services/language_translation/v2.js

```diff
@@ -152,29 +108,96 @@ module.exports = function (RED) {
         version: 'v2'
       });
 
-      var model_id = "";
-      if(domain === "news") {
+      switch (action) {
+        case 'translate':
+          this.doTranslate(msg);
+          break;
+        case 'train':
+          this.doTrain(msg);
+          break;
+        case 'getstatus':
+          this.doGetStatus(msg, trainid);
+          break;
+        case 'delete':
+          this.doDelete(msg, trainid);
+          break;
+      }
+    });
+
+    this.doTranslate = function (msg) {
+      var domain = msg.domain || config.domain;
+
+      if (!domain) {
+        node.warn('Missing translation domain, message not translated');
+        node.send(msg);
+        return;
+      }
+
+      var srclang  = ''; 
+      var destlang = '';
+
+      switch (domain) {
+        case 'news':
+          srclang = config.srclangnews;
+
+          if (!srclang) {
+            node.warn('Missing source language, message not translated');
+            node.send(msg);
+            return;
+          }
+
+          destlang = config.destlangnews;
+
+          if (!destlang) {
+            node.warn('Missing target language, message not translated');
+            node.send(msg);
+            return;
+          }
+          break;
+        case 'conversational':
+          srclang = config.srclangconversational;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 8 space characters but found 10.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2582431672/issues/source?bid=3225115&fileBranchId=3303298#l158)